src/reconcile/rec_visibility.c

```diff
@@ -292,6 +292,36 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
         if (WT_TXNID_LT(max_txn, txnid))
             max_txn = txnid;
 
+        /*
+         * If we're seeing out-of-order updates, we need to pin these updates into the cache.
+         *
+         * We're also clearing the selected update since nothing newer than this out-of-order update
+         * is allowed to be written to the disk. If we find something older in the update list that
+         * can be written to the disk, that's ok and we should go ahead and select it.
+         *
+         * If the previous update is globally visible, we don't need to care about out-of-order ids
+         * or timestamps since the offending update will be removed by an obsolete check further
+         * down the line.
+         */
+        if (!F_ISSET(r, WT_REC_IN_MEMORY) && prev_upd != NULL &&
```

Comment:
We can have U@5 -> aborted -> U@10 so I think this check is not enough.

Reply:
Yeah, good point. I should move the `prev_upd` assignment into the body and check for `WT_TXN_ABORTED` there.

Reply:
eb4054b